src/Microsoft.Azure.ServiceBus/Management/ManagementClient.cs

```diff
@@ -0,0 +1,588 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Azure.ServiceBus.Primitives;
+
+namespace Microsoft.Azure.ServiceBus.Management
+{
+    // TODO: Document all exceptions that might be thrown
+    // TODO: Retry for transient
+    public class ManagementClient : ClientEntity, IManagementClient
```

Comment:
What `ClientEntity` is for? If for connection, shouldn't it be incapsulated rather than ingerited?

Reply:
It gives you `IsClosedOrClosing`, `Connection`, `Timeout`, `RetryPolicy` and `CloseAsync()`. 

---
In reply to: [194202130](https://github.com/Azure/azure-service-bus-dotnet/pull/481#discussion_r194202130) [](ancestors = 194202130)

Reply:
Are those needed for `ManagementClient` internally? From the API standpoint it's not obvious that these are needed (looking at `IManagementClient`

Reply:
@nemakam it still feels off to implement `ClientEntity` rather than encapsulate one internally. Connection string would give you what you need, no?

Reply:
What part of that feels off?
Like I mentioned, we still need `IsClosedOrClosing`, and same thing with `OperationTimeout`, `RetryPolicy` and `CloseAsync()`.